go-controller/pkg/ovn/admin_network_policy_test.go

```diff
@@ -1025,6 +1077,473 @@ var _ = ginkgo.Describe("OVN ANP Operations", func() {
+					{
+						PortRange: &anpapi.PortRange{
+							Start:    36330,
+							End:      36550,
+							Protocol: v1.ProtocolUDP,
+						},
+					},
+					{
+						NamedPort: ptr.To("dns"), // matched by anpSubjectPod (will be created in the future)
+					},
+				}
+				anp := newANPObject("harry-potter", 65, anpSubject,
+					[]anpapi.AdminNetworkPolicyIngressRule{
+						{ // 3 ACLs + 1ACL for NamedPort
+							Name:   "deny-traffic-from-slytherin-to-gryffindor",
+							Action: anpapi.AdminNetworkPolicyRuleActionDeny,
+							From: []anpapi.AdminNetworkPolicyIngressPeer{
+								{
+									Namespaces: &anpapi.NamespacedPeer{
+										NamespaceSelector: &metav1.LabelSelector{
+											MatchLabels: peerDenyLabel,
+										},
+									},
+								},
+							},
+							Ports: &anpPortMangle,
+						},
+					},
+					[]anpapi.AdminNetworkPolicyEgressRule{
+						{ // 3 ACLs + 2ACLs for NamedPort
+							Name:   "pass-traffic-to-slytherin-from-gryffindor",
+							Action: anpapi.AdminNetworkPolicyRuleActionPass,
+							To: []anpapi.AdminNetworkPolicyEgressPeer{
+								{
+									Namespaces: &anpapi.NamespacedPeer{
+										NamespaceSelector: &metav1.LabelSelector{
+											MatchLabels: peerPassLabel,
+										},
+									},
+								},
+							},
+							Ports: &anpPortMangle,
+						},
+						{ // 3 ACLs
+							Name:   "allow-traffic-to-hufflepuff-from-gryffindor",
+							Action: anpapi.AdminNetworkPolicyRuleActionAllow,
+							To: []anpapi.AdminNetworkPolicyEgressPeer{
+								{
+									Namespaces: &anpapi.NamespacedPeer{
+										NamespaceSelector: &metav1.LabelSelector{
+											MatchLabels: peerAllowLabel,
+										},
+									},
+								},
+							},
+							Ports: &anpPortMangle,
+						},
+					},
+				)
+				anp.ResourceVersion = "1"
+				anp, err = fakeOVN.fakeClient.ANPClient.PolicyV1alpha1().AdminNetworkPolicies().Create(context.TODO(), anp, metav1.CreateOptions{})
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				acls := getACLsForANPRules(anp)
+				pg := getDefaultPGForANPSubject(anp.Name, []string{}, acls, false)
+				subjectNSASIPv4, subjectNSASIPv6 := buildNamespaceAddressSets(anpSubjectNamespaceName, []string{})
+				peerNSASIPv4, peerNSASIPv6 := buildNamespaceAddressSets(anpPeerNamespaceName, []string{anpPodV4IP2, anpPodV6IP2})
+				peerNS2ASIPv4, peerNS2ASIPv6 := buildNamespaceAddressSets(anpPeerNamespaceName+"2", []string{})
+				expectedDatabaseState := []libovsdbtest.TestData{pg, subjectNSASIPv4, subjectNSASIPv6, peerNSASIPv4, peerNSASIPv6, peerNS2ASIPv4, peerNS2ASIPv6}
+				expectedDatabaseState = append(expectedDatabaseState, getExpectedDataPodsAndSwitches([]testPod{t1}, []string{node1Name})...)
+				namedPortACLCount := 0
+				for _, acl := range acls {
+					acl := acl
+					if strings.Contains(acl.ExternalIDs[libovsdbops.PortPolicyProtocolKey.String()], "namedPort") {
+						namedPortACLCount++
+					}
+					expectedDatabaseState = append(expectedDatabaseState, acl)
+				}
```

Comment:
it was mainly for my self-OCD-satisfaction. Because we use `GetL3L4MatchesFromNamedPorts` from UTs and from main code; I wanted to make sure the logic used by `GetL3L4MatchesFromNamedPorts` was correct wrt to this specific test. So no specific need here, valid confusion its more for verbosity on test flow; let me add some comments explaining that so that others don't get the same confusion.

Reply:
this code seem to verify results from `getACLsForANPRulesWithNamedPorts`, so I would say it makes sense to create a separate test for `getACLsForANPRulesWithNamedPorts` in that case to ensure that correct acls are built, and then considering expectedDB is built properly, just compare it to the real db in this test.